packages/expo-ui/ios/Modifiers/ViewModifierRegistry.swift

```diff
@@ -5,6 +5,32 @@ import SwiftUI
 
 // MARK: - Individual ViewModifier Structs
 
+internal enum ListSectionSpacingType: String, Enumerable {
+  case `default`
+  case compact
+  case custom
+}
+
+internal struct ListSectionSpacingModifier: ViewModifier, Record {
+  @Field var spacing: ListSectionSpacingType = .default
+  @Field var value: CGFloat = 0
+
+  func body(content: Content) -> some View {
+    if #available(iOS 17.0, *) {
+      switch spacing {
+      case .compact:
+        content.listSectionSpacing(.compact)
+      case .custom:
+        content.listSectionSpacing(value)
+      default:
+        content.listSectionSpacing(.default)
+      }
+    } else {
+      content
+    }
+  }
+}
```

Comment:
```suggestion
internal struct ListSectionSpacingModifier: ViewModifier, Record {
  @Field var spacing: ListSectionSpacingType = .default
  @Field var value: CGFloat = 0

  func body(content: Content) -> some View {
    #if os(tvOS)
      content
    #else
      if #available(iOS 17.0, *) {
        switch spacing {
        case .compact:
          content.listSectionSpacing(.compact)
        case .custom:
          content.listSectionSpacing(value)
        default:
          content.listSectionSpacing(.default)
        }
      } else {
        content
      }
    #endif
  }
}
```

sorry my previous suggestion broke the tvos builds. this should fix it